src/main/java/com/minecolonies/coremod/entity/pathfinding/MinecoloniesAdvancedPathNavigate.java

```diff
@@ -888,8 +888,11 @@ public TreePathResult moveToTree(final BlockPos startRestriction, final BlockPos
         @NotNull final BlockPos start = AbstractPathJob.prepareStart(ourEntity);
         final BlockPos buildingPos = ((AbstractEntityCitizen) entity).getCitizenColonyHandler().getWorkBuilding().getPosition();
 
+        final BlockPos furthestRestriction = BlockPosUtil.getRandomPosInBoxPlane(BlockPosUtil
```

Comment:
whatfor do we need the furthest restriction?

Reply:
to calculate the max pathing range mostly.

Reply:
Since we search in the area from nearest to furthest so this somehow makes the heuristic work